doc/source/whatsnew/v2.0.0.rst

```diff
@@ -348,6 +348,7 @@ Other API changes
 - Passing a sequence containing a type that cannot be converted to :class:`Timedelta` to :func:`to_timedelta` or to the :class:`Series` or :class:`DataFrame` constructor with ``dtype="timedelta64[ns]"`` or to :class:`TimedeltaIndex` now raises ``TypeError`` instead of ``ValueError`` (:issue:`49525`)
 - Changed behavior of :class:`Index` constructor with sequence containing at least one ``NaT`` and everything else either ``None`` or ``NaN`` to infer ``datetime64[ns]`` dtype instead of ``object``, matching :class:`Series` behavior (:issue:`49340`)
 - :func:`read_stata` with parameter ``index_col`` set to ``None`` (the default) will now set the index on the returned :class:`DataFrame` to a :class:`RangeIndex` instead of a :class:`Int64Index` (:issue:`49745`)
+- Changed behavior of :class:`Index`, :class:`Series`, and :class:`DataFrame` arithmetic methods when working with object-dtypes, the results no longer do type inference on the result of the array operations (:issue:`49714`)
```

Comment:
So users would now call `convert_dtypes` to get close to the old behavior?

Might be good to mention because while this is more consistent behavior it's probably a little more inconvenient for users to work with object type

Reply:
infer_objects, will update

Reply:
i think adding this note will have to wait until 1) i add infer_objects to Index and possibly 2) a `copy=False` option is added to infer_objects

Reply:
Just merged the `infer_objects(copy=)` PR, does that need to be included here?

Reply:
ill add the `copy=False` in my next "assorted cleanups" branch